### DISCUSSION TITLE
Add tariff support into the energy module on Berry Tasmota scripting language

### DISCUSSION THREAD

**Discussion by tiagofreire-pt** [arendst/Tasmota#21922] [Ideas]
## Context
Use data from the tariffs previously setup to extract and process information for later on the webGUI, rules, etc.

## Need

- Read and write into the following new attributes

energy_tariff_1 = Returns or sets the total energy registered on the said N index tariff (float with 2 decimals)
energy_tariff_2 = ""
energy_tariff_3 = ""
energy_tariff_4 = ""
energy_tariff_5 = ""
energy_tariff_6 = ""
energy_tariff_7 = ""
energy_tariff_8 = ""
energy_tariff_9 = ""
current_tariff = Returns or sets the current tariff index (integer or 0 if tariffs disabled)

In: https://tasmota.github.io/docs/Berry/#energy-module

**Comment by s-hadinger** [comment]
Will take a look

**Comment by s-hadinger** [comment]
I looked at tariff, but it does not work as `tariff_1` .. `tariff_9`
At best you have `tarrif_1` and `tarrif_2` and they hold the `STD` and `DST` times. Then you also need `tariff_forced`.

Also `tariff` are just time in minutes, they are not float with 2 decimals.

What are you trying to achieve?

**Reply by Nazarbayevv** [reply]
Hello.

And what is "tariff_forced". I didn't seen it before on my old devices. But fount it nowadays on Shelly 3EM. And what is mean "Forced":"1"?

Because my device set all value to "Off-Peak".

{"Tariff":{"Off-Peak":{"STD":"23:00","DST":"23:00"},"Standard":{"STD":"07:00","DST":"07:00"},"Weekend":"OFF","Forced":"1"}}

**Reply by sfromis** [reply]
Looks like it is when you have used the command `Tariff 1`, which I think means "use off-peak tariff all weekend".

**Reply by Nazarbayevv** [reply]
How can I change this tariff forced parameter?

I didn’t find this in documents:(

**Reply by sfromis** [reply]
I already mentioned that it is a *command*, documented on the commands page.
https://tasmota.github.io/docs/Commands/#tariff

**Reply by Nazarbayevv** [reply]
Thanks, but this manual describe:
Tariff1  - Start off-peak tariff
Tariff2 - End off-peak tariff
Tariff9 -  on/off off-peak tariff for weekend. 

And nothing about "Forced":"1"

**Reply by sfromis** [reply]
I already very specifically referred to "use off-peak tariff all weekend". That is forcing it.

**Reply by Nazarbayevv** [reply]
I don't think so.

I recompiled firmware and "Forced" parameter was changed to "0". 

And i found this one on 262 line of tasmota_types.h:

    uint32_t tariff_forced : 2;            // bit 26/27 (v12.4.0.2) - Energy forced tariff : 0=tariff change on time, 1|2=tariff forced
asmota_types.h
https://github.com/arendst/Tasmota/blob/development/tasmota/include/tasmota_types.h

**Reply by sfromis** [reply]
That flag **is** what you can change using the `Tariff` command.
